jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/Server.java

```diff
@@ -278,10 +278,21 @@ private Server(
 		try {
 			jmxConnection = pool.borrowObject(this);
 			ImmutableList.Builder<Result> results = ImmutableList.builder();
-			MBeanServerConnection connection = jmxConnection.getMBeanServerConnection();
 
-			for (ObjectName queryName : query.queryNames(connection)) {
-				results.addAll(query.fetchResults(connection, queryName));
+			if(query.getQueryType() == Query.QueryType.POLL) {
```

Comment:
An Enum looks like a good candidate for a `switch` statement instead of `if`s.

Reply:
Sure... final cleanup pending :)